app/com/linkedin/drelephant/exceptions/azkaban/AzkabanJobLogExceptionFingerprinting.java

```diff
@@ -0,0 +1,85 @@
+package com.linkedin.drelephant.exceptions.azkaban;
+
+import com.linkedin.drelephant.clients.WorkflowClient;
+import com.linkedin.drelephant.configurations.scheduler.SchedulerConfigurationData;
+import com.linkedin.drelephant.exceptions.util.ExceptionInfo;
+import com.linkedin.drelephant.util.InfoExtractor;
+import com.linkedin.drelephant.util.Utils;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.log4j.Logger;
+
+import static com.linkedin.drelephant.exceptions.util.ExceptionUtils.ConfigurationBuilder.*;
+import static com.linkedin.drelephant.exceptions.util.ExceptionUtils.*;
+import static controllers.api.v1.JsonKeys.*;
+
+
+public class AzkabanJobLogExceptionFingerprinting {
+
+  private WorkflowClient _workflowClient;
+  private List<ExceptionInfo> azkabanExceptionInfoList = new ArrayList<>();
+  private final String SCHEDULER_NAME = "azkaban";
+  private final String PRIVATE_KEY = "private_key";
+
+  private static final Logger logger = Logger.getLogger(AzkabanJobLogExceptionFingerprinting.class);
+
+  public  AzkabanJobLogExceptionFingerprinting(String flowExecUrl, String jobExecUrl) {
+    logger.info("Initailizing AzkabanExceptionFingerPrinting class for " + jobExecUrl);
+    String jobId = getJobName(jobExecUrl);
+    _workflowClient = getWorkflowClient(flowExecUrl);
+    logger.info("Fetching job logs from Azkaban for " + jobExecUrl);
+    String jobLog = _workflowClient.getAzkabanJobLog(jobId, AZKABAN_JOB_LOG_START_OFFSET.getValue()
+            .toString(), AZKABAN_JOB_LOG_MAX_LENGTH.getValue().toString());
+    analyzeJobLog(jobLog);
```

Comment:
Why analyzeJobLog is part of constructor ?  It should be outside and should be called from client .  Currently constructor is doing lot of work , which is not correct as its breaching Single Responsibility